python/ray/dag/tests/experimental/test_execution_schedule.py

```diff
@@ -288,100 +288,116 @@ def test_two_ready_nccl_collectives(self, monkeypatch):
-    #     The following test case simulates a scenario where the READ
-    #     operations for x and y have already been added to the
-    #     execution schedule.
-    #     """
-    #     monkeypatch.setattr(ActorHandle, "__init__", mock_actor_handle_init)
-    #     fake_actor_1, dag_idx_1, local_idx_1 = ActorHandle("fake_actor_1"), 4, 2
-    #     fake_actor_2, dag_idx_2, local_idx_2 = ActorHandle("fake_actor_2"), 5, 1
-    #     dag_idx_3, local_idx_3 = 6, 3
-    #     mock_graph = {
-    #         dag_idx_1: generate_dag_graph_nodes(
-    #             local_idx_1, dag_idx_1, fake_actor_1, False, is_collective=True
-    #         ),
-    #         dag_idx_2: generate_dag_graph_nodes(
-    #             local_idx_2, dag_idx_2, fake_actor_2, False, is_collective=True
-    #         ),
-    #         dag_idx_3: generate_dag_graph_nodes(
-    #             local_idx_3, dag_idx_3, fake_actor_1, False, is_collective=True
-    #         ),
-    #     }
-    #     del mock_graph[dag_idx_1][_DAGNodeOperationType.READ]
-    #     del mock_graph[dag_idx_2][_DAGNodeOperationType.READ]
-
-    #     _add_edge(
-    #         mock_graph[dag_idx_1][_DAGNodeOperationType.COMPUTE],
-    #         mock_graph[dag_idx_1][_DAGNodeOperationType.WRITE],
-    #     )
-    #     _add_edge(
-    #         mock_graph[dag_idx_2][_DAGNodeOperationType.COMPUTE],
-    #         mock_graph[dag_idx_2][_DAGNodeOperationType.WRITE],
-    #     )
-    #     _add_edge(
-    #         mock_graph[dag_idx_3][_DAGNodeOperationType.COMPUTE],
-    #         mock_graph[dag_idx_3][_DAGNodeOperationType.WRITE],
-    #     )
-    #     _add_edge(
-    #         mock_graph[dag_idx_1][_DAGNodeOperationType.WRITE],
-    #         mock_graph[dag_idx_3][_DAGNodeOperationType.READ],
-    #     )
-    #     mock_actor_to_candidates = {
-    #         fake_actor_1: [mock_graph[dag_idx_1][_DAGNodeOperationType.COMPUTE]],
-    #         fake_actor_2: [mock_graph[dag_idx_2][_DAGNodeOperationType.COMPUTE]],
-    #     }
-    #     # Embed collective group information in _DAGNodeOperation.
-    #     mock_graph[dag_idx_2][_DAGNodeOperationType.COMPUTE].set_collective_edges(
-    #         {(dag_idx_3, _DAGNodeOperationType.COMPUTE)}
-    #     )
-    #     mock_graph[dag_idx_3][_DAGNodeOperationType.COMPUTE].set_collective_edges(
-    #         {(dag_idx_2, _DAGNodeOperationType.COMPUTE)}
-    #     )
-    #     next_nodes = _select_next_nodes(mock_actor_to_candidates, mock_graph)
-    #     assert len(next_nodes) == 1
-    #     assert next_nodes[0] == mock_graph[dag_idx_1][_DAGNodeOperationType.COMPUTE]
+    def test_nccl_collectives_one_ready_one_partial(self, monkeypatch):
+        """
+        Simulate the case where the only candidates are NCCL collective
+        operations. In this case, `_select_next_nodes` should return all
+        NCCL collective operations of the earliest-bound collective group
+        that is ready.
+
+        driver -> fake_actor_1.allreduce_1 -> driver
```

Comment:
Do you need to update the graph description here?

Reply:
done